lua/harpoon/init.lua

```diff
@@ -129,6 +132,7 @@ M.setup = function(config)
 
     local complete_config =
         merge_tables(
+            {global_settings = {}},
```

Comment:
Lets combine global_settings + projects table into one.

```
local complete_config =
        merge_tables(
            {projects = {}, global_settings = {["save_on_toggle"] = false}},
            ...
```

There is a bug here that will likely bite you.  You can save this by changing the idx from starting at 2 in merge_tables (i have a fix in a different branch).

This will make that option default off, but you can turn it on by calling setup!

Reply:
Also the change to merge tables should be like this

```
local function merge_tables(...)
    local out = {}
    for i = 1, select("#",...) do // IT WAS for i = 2, select("#",...) 
        merge_table_impl(out, select(i, ...))
    end
    return out
end
```

Reply:
something here goes wrong I think, it keeps initiating it as true. Feels like I'm missing something obvious, will push the test of the changes and take another look after dinner.